client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -254,7 +297,113 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
     label,
     rootSchema,
     renderers,
+    config,
   } = props;
+  const { errors: zErrors, options } = useZodOptionsValidation(
+    Options,
+    props.uischema.options
+  );
+
+  const defaultExpanded = props.uischema?.options?.['defaultExpanded'] ?? false;
+
+  const [expandedItems, setExpandedItems] = useState<boolean[]>(
+    new Array(inputData?.length ?? 0).fill(defaultExpanded)
+  );
+
+  const isNotesArray = options?.detail?.elements[0]?.type === 'Note';
+
+  const isElementEditable = (child: any, index: number) => {
+    if (!enabled) return false;
+    if (!options?.editRestrictions) return true;
+
+    const restrictions = options.editRestrictions;
+
+    // Must be editable if no timestamp (means it's just created)
+    if (!child.created) return true;
+
+    // Only allow the latest element to be edited
+    if (restrictions?.latest && index !== data.length - 1) return false;
+
+    // Author must match the current user
+    if (
+      restrictions?.isCurrentUser &&
+      child?.authorId &&
+      child.authorId !== config.user.id
+    )
+      return false;
+
+    // Must not be older than `maxAge` days
+    if (DateUtils.ageInDays(child.created) >= 1) return false;
+
+    return true;
+  };
+
+  const getItemLabel = (child: any, index: number) => {
+    const isExpanded = expandedItems[index];
+
+    // For most arrays, or Notes with and explicit itemLabel pattern
+    if (uischema.itemLabel || !isNotesArray)
+      return (
+        <Typography
+          sx={{
+            fontWeight: 'bold',
+            textAlign: 'end',
+            whiteSpace: 'nowrap',
+          }}
+        >
+          {uischema.itemLabel
+            ? RegexUtils.formatTemplateString(
+                uischema?.itemLabel,
+                {
+                  ...(typeof child === 'object' ? child : {}),
+                  index: index + 1,
+                },
+                ''
+              )
+            : index + 1}
+        </Typography>
+      );
+
+    // For Notes:
+    const {
+      text = '',
+      created,
+      authorName,
+    } = (inputData ? inputData[index] : {}) as {
+      text?: string;
+      created?: string;
+      authorName?: string;
+    };
+
+    return (
+      <div>
+        {!isExpanded ? (
+          <Typography
+            sx={{
+              textAlign: 'right',
+              whiteSpace: 'nowrap',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+              maxWidth: 350,
```

Comment:
I really wanted to not have to use an absolute pixel value here, but I couldn't get it to work with the overflow/ellipsis functionality. Without an absolute maximum, the text would just expand to the full width of what it *would* be if it weren't truncated. Very annoying. Would love it if we could figure this out.